engine/access/rpc/backend/backend.go

```diff
@@ -226,14 +225,51 @@ func executionNodesForBlockID(
 	state protocol.State) (flow.IdentityList, error) {
 
 	// lookup the receipts storage with the block ID
-	receipts, err := executionReceipts.ByBlockIDAllExecutionReceipts(blockID)
+	allReceipts, err := executionReceipts.ByBlockIDAllExecutionReceipts(blockID)
 	if err != nil {
 		return nil, fmt.Errorf("failed to retreive execution receipts for block ID %v: %w", blockID, err)
 	}
 
+	// execution result ID to execution receipt map to keep track of receipts by their result id
+	var identicalReceipts = make(map[flow.Identifier][]*flow.ExecutionReceipt)
+
+	// maximum number of matching receipts found so far for any execution result id
+	maxMatchedReceiptCnt := 0
+	// execution result id key for the highest number of matching receipts in the identicalReceipts map
+	var maxMatchedReceiptResultID flow.Identifier
+
+	// find the largest list of receipts which have the same result ID
+	for _, receipt := range allReceipts {
+
+		resultID := receipt.ExecutionResult.ID()
+		receipts, found := identicalReceipts[resultID]
+
+		if !found {
+			// first of its kind
+			receipts = []*flow.ExecutionReceipt{receipt}
+			identicalReceipts[resultID] = receipts
+			// keep looking for at least one more receipt with the same result
+			// since we need at least two execution receipt which have the same result
+			continue
+		}
+
+		// append receipt to the list of all receipts for this resultID
+		receipts = append(receipts, receipt)
+		identicalReceipts[resultID] = receipts
+
+		currentMatchedReceiptCnt := len(receipts)
+		if currentMatchedReceiptCnt > maxMatchedReceiptCnt {
+			maxMatchedReceiptCnt = currentMatchedReceiptCnt
+			maxMatchedReceiptResultID = resultID
+		}
+	}
+
+	// pick the largest list of matching receipts
+	matchingReceipts := identicalReceipts[maxMatchedReceiptResultID]
+
```

Comment:
We can also add a sanity check verifying the underlying assumption we're relying on here: that at most one EN encounters this failure for a given block:

```suggestion
if len(identicalReceipts) > 2 {
    // log an error
}
```

Reply:
you meant `< 2`? only one EN produced this receipt

Reply:
`identicalReceipts` is a mapping from result ID to the set of receipts which commit to that result ID. We are assuming that the worst case is one outlier EN which produces a faulty result ID and all other ENs agree (which results in `len(identicalReceipts) == 2`. If `len(identicalReceipts) > 2` it means there are 3 distinct result IDs which all disagree with each other.